pages/docs/manual/latest/function.mdx

```diff
@@ -415,6 +415,78 @@ If you write down the uncurried function's type, you'll add a dot there as well.
 
 **This feature seems trivial**, but is actually one of our most important features, as a primarily functional language. We encourage you to use it if you'd like to remove any mention of `Curry` runtime in the JS output.
 
+## Async/Await (from version v10.1)
+
+Just as in JS, an async function can be declared by adding `async` before the definition, and `await` can be used in the body of such functions.
+The outpul looks like idiomatic JS:
+
+<CodeTab labels={["ReScript", "JS Output"]}>
+
+```res example
+let getUserName = async (userId) => userId
+
+let greetUser = async (userId) => {
+  let name = await getUserName(userId)  
+  "Hello " ++ name ++ "!"
+}
+```
+```js
+async function greetUser(userId) {
+  var name = await getUserName(userId);
+  return "Hello " + name + "!";
+}
+```
+</CodeTab>
+
+The return type of `getUser` is inferred to be `promise<string>`, where type `promise` is now a built-in of the language.
```

Comment:
Not sure we need to mention that the promise type is now built in, might just confuse and it's an implementation detail that most users don't need to know about.